### DISCUSSION TITLE
How can I actually setup this project to contribute?

### DISCUSSION THREAD

**Discussion by ScriptPup** [mtxr/vscode-sqltools#1136] [Q&A] [answered]
I'm trying to follow the sparse steps listed on the [contribute page](https://vscode-sqltools.mteixeira.dev/en/contributing/building), but no matter what I try the `yarn` step simply fails when it gets to the node-sass build.

I've tried directly on my windows 10 system, I've tried in WSL Ubuntu-20.04, and I've tried on a VM with Ubuntu-22.04. They all fail. The windows one fails most spectacularly, not even getting to the node-sass build, but both of the Linux versions fail at the same spot.

```
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/scriptpup/projects/vscode-sqltools/build-tools/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:511:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
gyp ERR! System Linux 5.10.16.3-microsoft-standard-WSL2
gyp ERR! command "/usr/bin/node" "/home/scriptpup/projects/vscode-sqltools/build-tools/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /home/scriptpup/projects/vscode-sqltools/build-tools/node_modules/node-sass
gyp ERR! node -v v20.0.0
gyp ERR! node-gyp -v v8.4.1
gyp ERR! not ok
Build failed with error code: 1
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by gjsjohnmurray** [comment]
Thanks for being willing to try contributing.

Your output implies you're using Node v20. Please see the advice at https://vscode-sqltools.mteixeira.dev/en/contributing/building about this project requiring v14, and suggesting the use of `nvm` to accommodate this.

**Comment by ScriptPup** [comment]
Thanks for the response! 
However, I am using nvm and it is set to v14.
<img width="243" alt="image" src="https://user-images.githubusercontent.com/6163646/234022448-22f28c13-623d-41b0-ba54-254da36fc671.png">


I've never used `nvm` before, but it does seem to set the global version to whichever one I want - So I assume `yarn` should honor that. if there are any other steps though, I'm unaware of them. Is there something extra I need to do to make `yarn` use 14.17?

**Comment by gjsjohnmurray** [comment]
What yarn version do you have? I'm using v1.22.19 from https://classic.yarnpkg.com/lang/en/

**Comment by ScriptPup** [comment]
1.22.19 as well, I installed it using the Debian/Ubuntu instructions here https://classic.yarnpkg.com/en/docs/install#debian-stable - So I assume it should be the same thing. My `npm` version is 6.14.4. I don't see `gyp` installed, so I assume that's being installed as part of the `yarn` execution.

**Comment by ScriptPup** [comment, marked as answer]
Alright, I got it. Your questions helped give me the hint that `yarn` or the `node` version were probably the culprit.

I uninstalled all versions of node, uninstalled nvm, uninstalled npm, and uninstalled yarn (pretty much just nuked everything). After that I installed yarn first, then nvm, then installed version 14.21.3 (I noticed before I was using an older version of 14.x, so thought that might help. It didn't help by itself, I tried using this version via nvm previously). 

Once yarn, nvm, and only node 14.21 was installed I re-cloned the repo and ran `yarn`. This time it succeeded!

So I guess the answer is if you get an error similar to mine up above, just remove everything and try again without any potentially conflicting node or yarn versions installed?

It looks like everything is now working according to design. Thanks!

**Reply by gjsjohnmurray** [reply]
Thanks for persisting!

**Reply by gjsjohnmurray** [reply]
Please see https://github.com/mtxr/vscode-sqltools/discussions/1319 for the announcement that the project now uses node 16.

**Reply by Yahud100** [reply]
hii please help me , i am getting "There appears to be trouble with your network connection. Retrying... " and at end getting esockettimeout error , i have tried with wifi , lan and also with my mobile data , still same problem while doing lan , please help in this

**Reply by gjsjohnmurray** [reply]
@Yahud100 I saw you closed your https://github.com/mtxr/vscode-sqltools/discussions/1360 so I assume you no longer need help here.

**Comment by archiewood** [comment]
This is a very useful post, and does not show up in discussions automatically as it is marked as closed. Consider keeping this open?
